src/components/Table/TableHead/TableHeader.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings } from 'carbon-components';
+import {
+  ArrowUp20 as Arrow,
+  ArrowsVertical20 as Arrows,
+} from '@carbon/icons-react';
+
+import { sortStates } from './state/sorting';
+ 
+const { prefix } = settings;
+
+const translationKeys = {
+  iconDescription: 'carbon.table.header.icon.description',
+};
+
+const translateWithId = (key, { sortDirection, isSortHeader }) => {
+  if (key === translationKeys.iconDescription) {
+    if (isSortHeader) {
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Sort rows by this header in ascending order';
```

Comment:
I'm really confused on how translation is supposed to work here?  In many places the translation strings are defined in `i`8n` property... but here we seem to have a single `key` and then we use a complex set of if statements to turn that into hard coded text.   Is this a translation pattern that is used somewhere else?  I'm not sure how can translate this text?  @davidicus @tay1orjones  is a new iot translation pattern?